packages/sdk/src/multicall2.ts

```diff
@@ -0,0 +1,53 @@
+import { multicall2 } from "./clients";
+import { Call, Multicall, Request, State } from "./multicall";
+import { zip } from "lodash";
+import { SignerOrProvider } from ".";
+import { Contract } from "ethers";
+
+class Multicall2 extends Multicall<multicall2.Instance> {
+  constructor(state: State<multicall2.Instance>) {
+    super(state);
+  }
+
+  // reads from the contract, returns the read and error results in order that requests were queued.
+  public async readWithErrors(_requests: Request[] = this.requests) {
+    const encodedRequests = _requests.map((request) => this.encodeRequest(request));
+    const [, , returnData] = await this.multicallClient.callStatic.tryBlockAndAggregate(false, encodedRequests);
+    const zipped = zip(_requests, returnData);
+    return zipped.map(([request, response]) => {
+      if (request && response) {
+        return {
+          success: response.success,
+          result: response.success ? this.decodeResponse(request, response.returnData) : undefined,
```

Comment:
do you think we should copy the promise.allSettled return values?

```js
// [
//   {status: "fulfilled", value: 33},
//   {status: "fulfilled", value: 66},
//   {status: "fulfilled", value: 99},
//   {status: "rejected",  reason: Error: an error}
// ]
```

It doesnt change much except the actual props, just to standardize it a little.

Reply:
I like more boolean flags instead of string values. And multicall result doesn't include the error in case one of the calls fails